extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Key} from '../../../../build-system/tasks/e2e/e2e-types';
+
+describes.endtoend(
+  'amp-form [custom-validation-reporting=as-you-go]',
+  {
+    fixture: 'amp-form/validation-reporting-as-you-go.html',
+    environments: ['single'],
+  },
+  async (env) => {
+    let controller;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('should not send request when submitting default (invalid) input', async () => {
+      const submitButton = await controller.findElement('[type="submit"]');
+      await controller.click(submitButton);
+      await expect(
+        'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
+      ).to.not.have.been.sent;
```

Comment:
Curious, this check for a request being sent is natively built into our testing framework?

Reply:
Only on e2e tests.

```
extensions/amp-consent/0.1/test-e2e/test-amp-consent-client-side.js:
  102          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=sufficient'
  103:       ).to.have.been.sent;

extensions/amp-consent/0.1/test-e2e/test-amp-consent-server-side-expire-cache.js:
  47          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=insufficient'
  48:       ).to.have.been.sent;

  92          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=sufficient'
  93:       ).to.have.been.sent;

extensions/amp-consent/0.1/test-e2e/test-amp-consent-server-side.js:
  85          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking?consentState=insufficient'
  86:       ).to.have.been.sent;

extensions/amp-form/0.1/test-e2e/test-validation-reporting-as-you-go.js:
  36          'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  37:       ).to.not.have.been.sent;
  38      });

  48          'https://example.com/subscribe?__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  49:       ).to.have.been.sent;

extensions/amp-video/0.1/test-e2e/test-amp-video-analytics.js:
  53          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking&id=myVideo'
  54:       ).to.have.been.sent;

  64          'http://localhost:8000/amp4test/request-bank/e2e/deposit/tracking&id=myVideo2'
  65:       ).to.have.been.sent;

extensions/amp-video/0.1/test-e2e/test-amp-video-flexible-bitrate.js:
  228            'https://amp-dev.cdn.ampproject.org/mbv/s/amp.dev/static/samples/video/tokyo.mp4?amp_video_host_url=https%3A%2F%2Famp.dev%2F&__amp_source_origin=http%3A%2F%2Flocalhost%3A8000'
  229:         ).to.have.been.sent;
```